src/sagas/wallet.js

```diff
@@ -502,10 +508,42 @@ export function* onWalletReloadData() {
   yield put(startWalletSuccess());
 }
 
+export function* walletReloading() {
+  const wallet = yield select((state) => state.wallet);
+
+  // Since we close the channel after a walletReady event is received,
+  // we must fork this saga again so we setup listeners again.
+  yield fork(listenForWalletReady, wallet);
+
+  // Wait until the wallet is ready
+  yield take(types.WALLET_STATE_READY);
+
+  try {
+    const registeredTokens = yield call(loadTokens);
+
+    const customTokenUid = DEFAULT_TOKEN.uid;
+    const htrUid = hathorLibConstants.HATHOR_TOKEN_CONFIG.uid;
+    // We might have lost transactions during the reload, so we must invalidate the
+    // token histories:
```

Comment:
I would just add a comment here explaining why you skip the invalidation for HTR and default token. As far as I understood, the other reload wallet method already handles balance/history for them, right?

Also the names seem a bit confusing to me:

`onWalletReloadData` and `walletReloading`

Maybe this new one should be called `beforeWalletReload`. What do you think?

Reply:
Actually, we should activate this saga when `reload` is emitted from the facade as well. 

Refactored this in f7f70a20861de3573f1141cd21201b8aa02fa115

@r4mmer can you please review again?